falcon/request_helpers.py

```diff
@@ -129,3 +134,271 @@ def readlines(self, hint=None):
         """
 
         return self._read(hint, self.stream.readlines)
+
+
+class ParamProxy(object):
+    """
+    Provide `get_*` methods for both query-string parameters and
+    request body parameters.
+    """
+
+    def __init__(self, request, source):
+        self.req = request
+        self.source = source
+
+    def get(self, name, required=False, store=None):
```

Comment:
I like this idea. Good point about empty strings. We can get away with making default the first kwarg without introducing a breaking change, since it would be implemented only for the new attributes (i.e., `req.query` and `req.form`) while leaving the deprecated interface alone (i.e., `req.get_param()`)